contentgrid-appserver-rest/src/main/java/com/contentgrid/appserver/rest/problem/ContentGridExceptionHandler.java

```diff
@@ -220,6 +232,131 @@ ResponseEntity<Problem> handleEntityLinkedByRequiredRelation(@NonNull EntityLink
+                .withStatus(HttpStatus.NOT_FOUND)
+        );
+    }
+
+    @ExceptionHandler
+    ResponseEntity<Problem> handleEmptyRelation(@NonNull EmptyRelationException exception) {
+        var relationIdentity = exception.getRelationRequest();
+        var entityIdentity = EntityIdentity.forEntity(relationIdentity.getEntityName(), relationIdentity.getEntityId());
+
+        return createResponse(problemFactory.createProblem(ProblemType.NOT_FOUND_EMPTY_RELATION, relationIdentity.getRelationName(), entityIdentity)
+                .withStatus(HttpStatus.NOT_FOUND)
+        );
+    }
+
+    @ExceptionHandler
+    ResponseEntity<Problem> handleRelationTargetNotFound(@NonNull RelationTargetNotFoundException exception) {
+        var allExceptions = exception.allExceptions().toList();
+
+        var propertiesBuilder = ConstraintViolationProblemProperties.builder();
+        for (var ex : allExceptions) {
+            propertiesBuilder.target(
+                    handleEntityItemNotFound(ex).getBody(),
+                            ex.getEntityName().getValue(),
+                            ex.getId().getValue().toString()
+            );
+        }
+
+        var validationProblemProperties = propertiesBuilder.build();
+
+        return createResponse(
+                problemFactory.createProblem(ProblemType.INTEGRITY_INVALID_RELATION_TARGET)
+                        .withStatus(HttpStatus.BAD_REQUEST)
+                        .withProperties(validationProblemProperties)
+        );
+    }
+
+    @ExceptionHandler
+    ResponseEntity<Problem> handleInvalidRelationTarget(@NonNull InvalidRelationTargetException exception) {
+        return createResponse(
+                problemFactory.createProblem(ProblemType.INTEGRITY_INVALID_RELATION_TARGET)
+                        .withStatus(HttpStatus.BAD_REQUEST)
+                        .withProperties(properties -> properties.put("reference", exception.getReference()))
+        );
+    }
+
+    @ExceptionHandler
+    ResponseEntity<Problem> handleInvalidUriInList(@NonNull InvalidUriInListException exception) {
+        return createResponse(
+                problemFactory.createProblem(ProblemType.INTEGRITY_INVALID_RELATION_TARGET)
+                        .withStatus(HttpStatus.BAD_REQUEST)
+                        .withDetail(exception.getMessage())
+                        .withProperties(properties -> properties.put("reference", exception.getInvalid()))
+        );
+    }
+
+    @ExceptionHandler
+    ResponseEntity<Problem> handleResourceNotFound(@NonNull NoResourceFoundException exception) {
+        return createResponse(problemFactory.createProblem(ProblemType.NOT_FOUND_ENTITY_DEFINITION, exception.getResourcePath())
+                .withStatus(HttpStatus.NOT_FOUND)
+                .withProperties(Map.of("resource", exception.getResourcePath()))
+        );
+    }
```

Comment:
I would not handle this one at all.

I you're going to send "entity or resource not found", it is not really any improvement over sending a plain 404.